sqlglot/generator.py

```diff
@@ -2860,16 +2860,27 @@ def ignorenulls_sql(self, expression: exp.IgnoreNulls) -> str:
     def respectnulls_sql(self, expression: exp.RespectNulls) -> str:
         return self._embed_ignore_nulls(expression, "RESPECT NULLS")
 
+    def havingmax_sql(self, expression: exp.HavingMax) -> str:
+        this_sql = self.sql(expression, "this")
+        expression_sql = self.sql(expression, "expression")
+        kind = "MAX" if expression.args.get("max") else "MIN"
+        return f"{this_sql} HAVING {kind} {expression_sql}"
+
     def _embed_ignore_nulls(self, expression: exp.IgnoreNulls | exp.RespectNulls, text: str) -> str:
         if self.IGNORE_NULLS_IN_FUNC and not expression.meta.get("inline"):
-            for klass in (exp.Order, exp.Limit):
-                mod = expression.find(klass)
-
-                if mod:
-                    this = expression.__class__(this=mod.this.copy())
-                    this.meta["inline"] = True
-                    mod.this.replace(this)
-                    return self.sql(expression.this)
+            mods = sorted(
```

Comment:
I think I'd add a comment like this here because it's non-trivial; ou have to know the parser control flow to understand the sort order
```suggestion
            # The first modifier here will be the one closest to the AggFunc's arg
            mods = sorted(
```

Reply:
why not do a for loop like previously?

Reply:
it’s more expensive because you have to search n times

Reply:
ok, do you wanna go with the comment or leave it as-is? i think that's the only comment, looks good otherwise